include/picongpu/fields/laserProfiles/laserExpRampWithPrepulse.hpp

```diff
@@ -0,0 +1,189 @@
+/* Copyright 2013-2017 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch, Stefan Tietze
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+
+namespace picongpu
+{
+/** Wavepacket with spatial gaussian envelope and adjustable temporal shape.
+ * Allows defining a prepulse and two regions of exponential preramp with
+ * independent slopes. The definition works by specifying three (t, intensity)-
+ * points, where time is counted from the very beginning in SI and the
+ * intensity (yes, intensity, not amplitude) is given in multiples of the main
+ * peak.
+ *
+ * Be careful - problematic for few cycle pulses. Thought the rest is cloned
+ * from laserWavepacket, the correctionFactor is not included (this made a
+ * correction to the laser phase, which is necessary for very short pulses,
+ * since otherwise the field is not a solution to the the free Maxwell Eq.; so
+ * that in particular a test particle is, after the laser pulse has passed,
+ * not returned to immobility, as it should. Since the analytical solution is
+ * only implemented for the gaussian regime, and we have mostly exponential
+ * regimes here, it was not retained here.
+ */
+
+namespace laserExpRampWithPrepulse
+{
+    constexpr float_X laserTimeShift = laser::initPlaneY * CELL_HEIGHT /
+        SPEED_OF_LIGHT;
+    constexpr float_64 f = SPEED_OF_LIGHT / WAVE_LENGTH;
```

Comment:
please check if 64 bit is required

Reply:
It seems to be required, since in all the other laser profiles there are also float_64 for anything connected to time, and float_X for the other, amplitude-connected values. 
Probably a nontrivial question, and probably someone thought about it before without documenting it.